cryptoconditions/types/prefix.py

```diff
@@ -0,0 +1,274 @@
+        prefixed message.
+
+        Note that you only have to add either the subcondition or a
+        subfulfillment, but not both.
+
+        Args:
+            subfulfillment {Fulfillment|str}:  Fulfillment object or URI
+                string representing the fulfillment to use as the
+                subcondition.
+
+        """
+        if isinstance(subfulfillment, str):
+            subfulfillment = Fulfillment.from_uri(subfulfillment)
+        elif not isinstance(subfulfillment, Fulfillment):
+            raise Exception(
+                'Subfulfillments must be URIs or objects of type Fulfillment')
+        self._subcondition = subfulfillment
+
+    @property
+    def prefix(self):
+        """Set the prefix.
+
+        The prefix will be prepended to the message during validation
+        before the message is passed on to the subcondition.
+
+        Args:
+            prefix (bytes): Prefix to apply to the message.
+
+        """
+        return self._prefix
+
+    @prefix.setter
+    def prefix(self, prefix):
+        if not isinstance(prefix, bytes):
+            raise TypeError('Prefix must be a Buffer, was: {}'.format(prefix))
+        self._prefix = prefix
+
+    @property
+    def max_message_length(self):
+        return self._max_message_length
+
+    @max_message_length.setter
+    def max_message_length(self, max_message_length):
+        """The threshold.
+
+        Determines the threshold that is used to consider this
+        condition fulfilled. If the number of valid subfulfillments is
+        greater or equal to this number, the threshold condition is
+        considered to be fulfilled.
+
+        Args:
+            max_message_length (int): Threshold.
+
+        """
+        if not isinstance(max_message_length, int) or max_message_length < 0:
+            raise TypeError(
+                'Max message length must be an integer greater than or '
+                'equal to zero, was: {}'.format(max_message_length)
+            )
+
+        self._max_message_length = max_message_length
+
+    @property
+    def subtypes(self):
+        """Get types used in this condition.
+
+         This is a type of condition that contains a subcondition. A
+         complete set of subtypes must contain the set of types that must
+         be supported in order to validate this fulfillment. Therefore,
+         we need to join the type of this condition to the types used in
+         the subcondition.
+
+         Returns:
+             :obj:`set` of :obj:`str`: Complete type names for this
+                 fulfillment.
+
+        Note:
+            Never include our own type as a subtype. The reason is that
+            we already know that the validating implementation knows how
+            to interpret this type, otherwise it wouldn't be able to
+            verify this fulfillment to begin with.
+
+        """
+        return {t for t in chain(self.subcondition.subtypes,
```

Comment:
if I am not wrong, the `subcondition.subtypes` should list all subtypes, regardless of the nesting level